lib/immutable-struct.rb

```diff
@@ -72,6 +72,15 @@ def self.new(*attributes,&block)
         end
       end
 
+      define_singleton_method(:from) do |value|
+        case value
+        when self then value
+        when Hash then new(value)
```

Comment:
If this is coercion, should it call `to_hash` on the incoming object?

Reply:
I want to say "no",  because we're not trying to coerce the value into a hash but rather into a "whatever this immutable struct is". Even if the given value were convertible to Hash it would have to be a very specific hash form. Does that make sense?